_studio/mfx_lib/shared/include/mfx_h264_encode_struct_vaapi.h

```diff
@@ -314,6 +314,20 @@ typedef struct tagENCODE_CAPS
     UCHAR    MaxNum_WeightedPredL1;
 } ENCODE_CAPS;
 
+// We cannot modify ENCODE_CAPS structure, but need to add new features
+struct MFX_ENCODE_CAPS
+{
+    ENCODE_CAPS ddi_caps;
```

Comment:
why not inherit new caps from previous? it will remove all those massive changes to introduce ```.ddi_caps``` in myriad places.
```
struct MFX_ENCODE_CAPS : ENCODE_CAPS
{
    union {
        struct {
            UINT CQPBRCSupport: 1;
            UINT CBRSupport   : 1;
            UINT VBRSupport   : 1;
            UINT              : 28;
        };
        UCHAR   CodingLimits3;
    }
};
```

Reply:
First, i thought about inherit, but @dmitryermilov said me what current way more better.

Reply:
ok

Reply:
My way provide more clearly interface and separates our structure from ddi. 
But your way is better from another point, because this have no need to massive refactor.
What we must choose? Dmitry doesn`t mind about inheritance, but from my point of view it`s a matter of taste.

Reply:
I'm personally ok with both: inheritance and inclusion. Advantage of inclusion is it will be clear where ddi part of caps is and where our custom part is.

Reply:
I would vote for inheritance. Less code, much less changes, more modern style, which was introduced for such cases.

Reply:
I think we can do following thing:
1) Use inheritance
2) Pass caps by reference using ```ENCODE_CAPS const & hwCaps``` object everywhere when possible
3) Use ```MFX_ENCODE_CAPS``` only when required

@dmitryermilov what do you think?

Reply:
Hi @mgonchar , 
Sorry but I don't see the idea behind 2) and 3) above.  The purpose of this change is to extend current ENCODE_CAPS structure. So apparently once we extend it, we need to use a new MFX_ENCODE_CAPS (which encapsulates old  ENCODE_CAPS) everywhere.

Reply:
Idea is to not introduce sufficient delta. I.e. most of the functions accepting caps don't need our additional fields, and if we use const ref and pass base class (even if we pass only .ddi in new notation) it will decrease delta of this patch + limit scope of usage of new fields. Honestly I just don't like huge delta which we have now and thinking about reducing it, I think we can do it even if extension will be done via inclusion

Reply:
For 2-3) look for example here: https://github.com/Intel-Media-SDK/MediaSDK/pull/1432/files#diff-0bd236db445515bb8ccdb4a89eb6f852L220 I think that better will be to not change signature of this function, but pass ```.ddi_caps``` member inside instead. @onabiull @dmitryermilov what do you think about such simplification?

Reply:
I do see your point. However I think we shouldn't be afraid of changing code here. IMO, confusion , which readers may have coming across different caps strictures is more important. Just my private opinion.

Reply:
ok. I removed my -1 for this one since no major issues remained here